### PR TITLE
Feat: environment Role assignments for Team

### LINKED ISSUE #654: Environment Role assignments for User/Team
Terraform Provider **resource**/**data** for the new Environment Role Assignments

Expected Result / DoD ✅:

Terraform Provider resource/data for the new "Environment Role Assignments".

create new OR use an existing resource we have, but just use the new APIs while demand from the user to give environmentId instead of projectId)

There are 2 different assignments - User / Team

Use the new [APIs](https://docs.env0.com/reference/roles-get-user-role-assignments): Get, Upsert, Delete

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
resolves #654 

### Solution

1. Added new API calls.
2. Added unit tests for API calls.
3. Created a new team environment assignment resource.
4. Added acceptance tests.
5. Added examples.
6. Added integration tests.

### CODE REVIEW

**Comment by TomerHeber** [comment]
Important note:

This is very similar to the user environment assignment (pretty much a copy-paste).
If another one comes along (an assignment) that is similar I'll merge it all together into a common code.
I can do it now as well if preferred.

File: examples/resources/env0_team_environment_assignment/resource.tf [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+resource "env0_team" "team" {
+  name = "team"
+}
+
+resource "env0_custom_role" "custom_role" {
+  name = "custom-role-sample"
+  permissions = [
+    "VIEW_PROJECT",
```
Comment (yarivg) [inline examples/resources/env0_team_environment_assignment/resource.tf:22]: change permissions to be ENVIRONMENT RELATED.. like view-env, edit-env-settings

File: examples/resources/env0_workflow_triggers copy/resource.tf [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+resource "env0_api_key" "api_key" {
+  name              = "user-api-key-sample"
+  organization_role = "User"
+}
+
+resource "env0_custom_role" "custom_role" {
+  name = "custom-role-sample"
+  permissions = [
+    "VIEW_PROJECT",
```
Comment (yarivg) [inline examples/resources/env0_workflow_triggers copy/resource.tf:23]: change permissions to be ENVIRONMENT RELATED.. like view-env, edit-env-settings

**Review by yarivg — APPROVED** [review]
Seems good!

File: examples/resources/env0_team_environment_assignment/resource.tf (hunk 1 above)
Comment (TomerHeber) [reply examples/resources/env0_team_environment_assignment/resource.tf:22]: Hi @yarivg - could you please explain this comment?

**Pull request merged**